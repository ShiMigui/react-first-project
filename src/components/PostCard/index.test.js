import { render, screen } from "@testing-library/react";
import { PostCard } from '.';

const mock = {
    title: 'Hello World',
    body: 'This is a test',
    url: 'https://www.somesite.com/img.jpg',
}

const _component = <PostCard {...mock} />;

describe('<PostCard />', () => {
    it('must be in the document', () => {
        render(_component);

        let el = screen.getByRole('img', { name: mock.title });
        expect(el).toBeInTheDocument();
        expect(el).toHaveAttribute('src', mock.url);

        el = screen.getByRole('heading', { name: mock.title });
        expect(el).toBeInTheDocument();

        el = screen.getByText(mock.body);
        expect(el).toBeInTheDocument();
    })

    it('must match with the snapshot', () => {
        render(_component);

        expect(screen.getByTestId("post-card")).toMatchSnapshot();
    })
})