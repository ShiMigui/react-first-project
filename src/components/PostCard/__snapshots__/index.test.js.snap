// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> must match with the snapshot 1`] = `
<div
  class="post-card"
  data-testid="post-card"
>
  <img
    alt="Hello World"
    src="https://www.somesite.com/img.jpg"
  />
  <div
    class="post-content"
  >
    <h4>
      Hello World
    </h4>
    <p>
      This is a test
    </p>
  </div>
</div>
`;
